__tests__/app.test.js

```diff
@@ -192,195 +192,373 @@ describe("/api/articles", () => {
+            comments.forEach((comment) => {
+              expect(comment).toHaveProperty("comment_id");
+              expect(comment).toHaveProperty("article_id");
+              expect(comment).toHaveProperty("votes");
+              expect(comment).toHaveProperty("created_at");
+              expect(comment).toHaveProperty("author");
+              expect(comment).toHaveProperty("body");
+            });
+          });
+      });
+      test("400: returns appropriate error message when given invalid id", () => {
+        return request(app)
+          .get("/api/articles/tree/comments")
+          .expect(400)
+          .then((response) => {
+            expect(response.body.msg).toBe("Bad request");
+          });
+      });
+      test("404: responds with appropriate error message when id does not exist", () => {
+        return request(app)
+          .get("/api/articles/1231212/comments")
+          .expect(404)
+          .then((response) => {
+            expect(response.body.msg).toBe("Article id does not exist");
+          });
+      });
+      test("200: responds with empty array when id does exist but no results are returned", () => {
+        return request(app)
+          .get("/api/articles/2/comments")
+          .expect(200)
+          .then((response) => {
+            const { comments } = response.body;
+            expect(comments).toEqual([]);
+          });
+      });
+    });
+    describe("POST", () => {
+      test("201: creates new comment and responds with newly created comment from table", () => {
+        return request(app)
+          .post("/api/articles/3/comments")
+          .send({ username: "rogersop", body: "A whacky good read" })
+          .expect(201)
+          .then((response) => {
+            const { comment } = response.body;
+            expect(comment).toHaveProperty("votes", 0);
+            expect(comment).toHaveProperty("comment_id");
+            expect(comment).toHaveProperty("created_at");
+            expect(comment).toHaveProperty("body", "A whacky good read");
+            expect(comment).toHaveProperty("article_id", 3);
+            expect(comment).toHaveProperty("author", "rogersop");
+          });
+      });
+      test("400: responds with appropriate error message when invalid id is sumbimtted", () => {
+        return request(app)
+          .post("/api/articles/tree/comments")
+          .send({ username: "rogersop", body: "A whacky good read" })
+          .expect(400)
+          .then((response) => {
+            expect(response.body.msg).toBe("Bad request");
+          });
+      });
+      test("400: responds with appropriate error message when id does not exist", () => {
+        return request(app)
+          .post("/api/articles/123234325/comments")
+          .send({ username: "rogersop", body: "A whacky good read" })
+          .expect(400)
+          .then((response) => {
+            expect(response.body.msg).toBe("Bad request");
+          });
+      });
```

Comment:
looks like this error should instead be a 404 - the request is valid apart from a non existent articleID.